dask_glm/algorithms.py

```diff
@@ -181,17 +186,16 @@ def newton(X, y, max_iter=50, tol=1e-8, family=Logistic, **kwargs):
         step, _, _, _ = np.linalg.lstsq(hess, grad)
         beta = (beta_old - step)
 
-        iter_count += 1
-
         # should change this criterion
         coef_change = np.absolute(beta_old - beta)
         converged = (
-            (not np.any(coef_change > tol)) or (iter_count > max_iter))
+            (not np.any(coef_change > tol)) or (n_iter >= max_iter))
 
         if not converged:
             Xbeta = dot(X, beta)  # numpy -> dask converstion of beta
+            n_iter += 1
```

Comment:
Should `n_iter += 1` be moved to the line after `beta = (beta_old - step)`? Feel free to correct me if I'm wrong, but I think that in the limiting case that the convergence condition is met after a single iteration, this will return `n_iter = 0` right now

Reply:
Hmm, but would that be consider an iteration if no work has been done (e.g., convergence was achieved without work)? I honestly didn't consider that this could happen.

Reply:
I think that at least one iteration is always done when this function is called. `n_iter` starts out as 0 with `converged = False`

https://github.com/dask/dask-glm/blob/34766b23bfec90a51e7afeaeddfce6d9f096957c/dask_glm/algorithms.py#L172-L173

then we enter the `while` loop and work is done to get an updated value for `beta`

https://github.com/dask/dask-glm/blob/34766b23bfec90a51e7afeaeddfce6d9f096957c/dask_glm/algorithms.py#L182-L187

We then check to see if things have converged 

https://github.com/dask/dask-glm/blob/34766b23bfec90a51e7afeaeddfce6d9f096957c/dask_glm/algorithms.py#L190-L192

Let's pretend that the first time line 191 is evaluated we've converged and now `converged = True`. Now we won't go into the code block that increments `n_iter`

https://github.com/dask/dask-glm/blob/34766b23bfec90a51e7afeaeddfce6d9f096957c/dask_glm/algorithms.py#L194-L196

then we'll break out of the `while` loop and `n_iter` will be returned as `0` even though we've already done one iteration. `n_iter` being off by one should happen for any number of iterations (e.g. `n_iter` being returned as 5 after 6 iterations), it's just easier for me to think about the single iteration case.

Reply:
Ok, I see what you mean and I think you're right, my bad. Fixing.